sdk/search/search-documents/review/search-documents.api.md

```diff
@@ -2065,14 +2066,14 @@ export interface SearchClientOptions extends ExtendedCommonClientOptions {
 }
 
 // @public
-export interface SearchDocumentsPageResult<T> extends SearchDocumentsResultBase {
+export interface SearchDocumentsPageResult<Model extends object, Fields extends SelectFields<Model>> extends SearchDocumentsResultBase {
     continuationToken?: string;
-    readonly results: SearchResult<T>[];
+    readonly results: SearchResult<Model, Fields>[];
 }
 
 // @public
-export interface SearchDocumentsResult<T> extends SearchDocumentsResultBase {
-    readonly results: SearchIterator<T>;
+export interface SearchDocumentsResult<Model extends object, Fields extends SelectFields<Model>> extends SearchDocumentsResultBase {
```

Comment:
Using that default type erases type narrowing, but it won't break. I think it makes sense to add a generic parameter if you're declaring the type manually, but it shouldn't be required. I've put the default `Fields` value in, and will make sure the rest of the API matches.

Reply:
To me, as long as the types can be inferred strongly, as much as possible, from the _method call_, then I think that leads to the best experience for customers.

Reply:
It's a bit tricky to declare the correct narrowed type, but the method call doesn't require any of the type parameters to be declared explicitly to get the most narrow type possible, so I'm happy to call this as good as it can get.